src/Raven.Client/Documents/Session/TimeSeries/TimeSeriesEntry.cs

```diff
@@ -122,4 +182,230 @@ void IPostJsonDeserialization.PostDeserialization()
+
+                Build2DArray();
+                _max = new TTimeSeriesEntry
+                {
+                    Timestamp = Timestamp, 
+                    Values = _innerValues[(int)AggregationType.Max]
+                };
+                _max.SetMembersFromValues();
+                return _max;
+            }
+        }
+
+        [JsonIgnore]
+        public TTimeSeriesEntry Sum
+        {
+            get
+            {
+                if (_sum?.Values != null)
+                    return _sum;
+
+                Build2DArray();
+                _sum = new TTimeSeriesEntry
+                {
+                    Timestamp = Timestamp, 
+                    Values = _innerValues[(int)AggregationType.Sum]
+                };
+                _sum.SetMembersFromValues();
+                return _sum;
+            }
+        }
+
+        [JsonIgnore]
+        public TTimeSeriesEntry Count
+        {
+            get
+            {
+                if (_count?.Values != null)
+                    return _count;
+
+                Build2DArray();
+                _count = new TTimeSeriesEntry
+                {
+                    Timestamp = Timestamp,
+                    Values = _innerValues[(int)AggregationType.Count]
+                };
+                _count.SetMembersFromValues();
+                return _count;
+            }
+        }
+
+        [JsonIgnore]
+        public TTimeSeriesEntry Average
+        {
+            get
+            {
+                if (_average?.Values != null)
+                    return _average;
+
+                Build2DArray();
+                var arr = new double[_dim];
+                for (int i = 0; i < arr.Length; i++)
+                {
+                    arr[i] = _innerValues[(int)AggregationType.Sum][i] / _innerValues[(int)AggregationType.Count][i];
+                }
+
+                _average = new TTimeSeriesEntry
+                {
+                    Timestamp = Timestamp,
+                    Values = arr
+                };
+                _average.SetMembersFromValues();
+                return _average;
+            }
+        }
+
+        private bool _innerArrayInitialized;
+        private void Build2DArray()
+        {
+            if (IsRollup == false)
+                throw new InvalidOperationException("Not a rolled up entry.");
+
+            if (_innerArrayInitialized)
+                return;
+
+            _dim = Values.Length / 6;
+            _innerValues = new double[6][];
+            for (int i = 0; i < 6; i++)
+            {
+                _innerValues[i] = new double[_dim];
+                for (int j = 0; j < _dim; j++)
+                {
+                    _innerValues[i][j] = Values[j * 6 + i];
+                }
+            }
+
+            _innerArrayInitialized = true;
+        }
+
+        private double[][] _innerValues;
+
+        [OnDeserialized]
+        internal new void OnNewtonSoftJsonDeserialized(StreamingContext context)
```

Comment:
@karmeli87 I just realized that after https://github.com/ravendb/ravendb/pull/10580 is merged I would like to avoid this kind of an approach. I think it would be better if you would write converter (supporting reads only).